Add did:nuggets method

amlord opened the pull request:
### DID Method Registration

As a DID method registrant, I have ensured that my DID method registration complies with the following statements:

- [x] The DID Method specification [defines the DID Method Syntax](https://w3c.github.io/did-core/#method-syntax).
- [x] The DID Method specification [defines the Create, Read, Update, and Deactivate DID Method Operations](https://w3c.github.io/did-core/#method-operations).
- [x] The DID Method specification [contains a Security Considerations section](https://w3c.github.io/did-core/#security-requirements).
- [x] The DID Method specification [contains a Privacy Considerations section](https://w3c.github.io/did-core/#privacy-requirements).
- [x] The JSON file I am submitting has [passed all automated validation tests below](#partial-pull-merging).
- [x] The JSON file contains a `contactEmail` address [OPTIONAL].
- [x] The JSON file contains a `verifiableDataRegistry` entry [OPTIONAL].

msporny reviewed (approved):
The DID Method specification meets all requirements needed for provisional registration.

kdenhartog reviewed (approved):
Approving because all requirements are met at this point and this is the consistent way to handle registrations. With that in mind, I found a few implementation concerns that I had and I'll follow up with issues raised on that repo on how you can further improve the method for implementation and interoperability.

amlord commented:
> Approving because all requirements are met at this point and this is the consistent way to handle registrations. With that in mind, I found a few implementation concerns that I had and I'll follow up with issues raised on that repo on how you can further improve the method for implementation and interoperability.

that'd be great - thanks @kdenhartog!

OR13 reviewed (changes requested):
<img width="905" alt="Screen Shot 2021-12-21 at 6 43 12 PM" src="https://user-images.githubusercontent.com/8295856/147015853-d5588174-7e30-402a-990d-d1eb32beb1e4.png">


@peacekeeper would you please confirm this did method privacy section is acceptable?

peacekeeper commented:
> @peacekeeper would you please confirm this did method privacy section is acceptable?

Since the method spec mentions a "private Ethereum network" and an "in-app onboarding KYC process", I think it would be helpful to add some more information on what this means for the privacy of the DID subject, e.g. who operates the resolver endpoint(s), and what data is being collected about the DID subject in the KYC process..

amlord commented:
> > @peacekeeper would you please confirm this did method privacy section is acceptable?
> 
> Since the method spec mentions a "private Ethereum network" and an "in-app onboarding KYC process", I think it would be helpful to add some more information on what this means for the privacy of the DID subject, e.g. who operates the resolver endpoint(s), and what data is being collected about the DID subject in the KYC process..

Hi @peacekeeper, thanks for your input - I've raised a PR in the other repo, which I hope addresses your concerns: https://github.com/NuggetsLtd/nuggets-did-resolver/pull/9

amlord commented:
@kdenhartog - thanks for the issues you've raised on the resolver repo; these are greatly received & haven't been ignored - I'll be updating to reflect these items in the new year, once I'm back from annual leave...

amlord commented:
Happy New Year everyone! @peacekeeper @OR13 the PR updated privacy considerations has now been merged into the main branch of the repo - does this now address the concerns raised?

amlord commented:
Good morming @peacekeeper @OR13 @kdenhartog @msporny @mprorock, it's been nearly 3 weeks since changes were requested on this PR (which I addressed the same day). Since this time there's not been a response; I appreciate that it's been holiday season, but I've seen comments from reviewers I'm waiting on for other PRs to this repo...

If we could get some focus on this, that'd be great - thanks!

peacekeeper reviewed (approved):
@amlord sorry for the delay in getting back to you. I think the additions you made to the Privacy Considerations are really good, and this should be merged!

amlord commented:
@peacekeeper awesome, thanks!

msporny commented:
I just re-ping'd @OR13 to do a review. We just need his approval and we can merge.

amlord commented:
@msporny OR13 has approved - is this all that's needed to merge now? Thanks...

kdenhartog commented:
This has been opened for a sufficient period of time, no objections have been noted and all editors have approved this method registration. Merging.
